src/execution/vm/vm.cpp

```diff
@@ -0,0 +1,1737 @@
+    auto top_k = frame->LocalAt<u64>(READ_LOCAL_ID());
+    OpSorterSortTopKParallel(sorter, thread_state_container, sorter_offset, top_k);
+    DISPATCH_NEXT();
+  }
+
+  OP(SorterFree) : {
+    auto *sorter = frame->LocalAt<sql::Sorter *>(READ_LOCAL_ID());
+    OpSorterFree(sorter);
+    DISPATCH_NEXT();
+  }
+
+  OP(SorterIteratorInit) : {
+    auto *iter = frame->LocalAt<sql::SorterIterator *>(READ_LOCAL_ID());
+    auto *sorter = frame->LocalAt<sql::Sorter *>(READ_LOCAL_ID());
+    OpSorterIteratorInit(iter, sorter);
+    DISPATCH_NEXT();
+  }
+
+  OP(SorterIteratorHasNext) : {
+    auto *has_more = frame->LocalAt<bool *>(READ_LOCAL_ID());
+    auto *iter = frame->LocalAt<sql::SorterIterator *>(READ_LOCAL_ID());
+    OpSorterIteratorHasNext(has_more, iter);
+    DISPATCH_NEXT();
+  }
+
+  OP(SorterIteratorNext) : {
+    auto *iter = frame->LocalAt<sql::SorterIterator *>(READ_LOCAL_ID());
+    OpSorterIteratorNext(iter);
+    DISPATCH_NEXT();
+  }
+
+  OP(SorterIteratorGetRow) : {
+    const auto **row = frame->LocalAt<const byte **>(READ_LOCAL_ID());
+    auto *iter = frame->LocalAt<sql::SorterIterator *>(READ_LOCAL_ID());
+    OpSorterIteratorGetRow(row, iter);
+    DISPATCH_NEXT();
+  }
+
+  OP(SorterIteratorFree) : {
+    auto *iter = frame->LocalAt<sql::SorterIterator *>(READ_LOCAL_ID());
+    OpSorterIteratorFree(iter);
+    DISPATCH_NEXT();
+  }
+
+  // -------------------------------------------------------
+  // Output Calls
+  // -------------------------------------------------------
+  OP(OutputAlloc) : {
+    auto exec_ctx = frame->LocalAt<exec::ExecutionContext *>(READ_LOCAL_ID());
+    auto *result = frame->LocalAt<byte **>(READ_LOCAL_ID());
+    OpOutputAlloc(exec_ctx, result);
+    DISPATCH_NEXT();
+  }
+
+  OP(OutputAdvance) : {
+    auto exec_ctx = frame->LocalAt<exec::ExecutionContext *>(READ_LOCAL_ID());
+    OpOutputAdvance(exec_ctx);
+    DISPATCH_NEXT();
+  }
+
+  OP(OutputSetNull) : {
+    auto exec_ctx = frame->LocalAt<exec::ExecutionContext *>(READ_LOCAL_ID());
+    auto idx = frame->LocalAt<u32>(READ_LOCAL_ID());
+    OpOutputSetNull(exec_ctx, idx);
+    DISPATCH_NEXT();
+  }
+
+  OP(OutputFinalize) : {
+    auto exec_ctx = frame->LocalAt<exec::ExecutionContext *>(READ_LOCAL_ID());
+    OpOutputFinalize(exec_ctx);
+    DISPATCH_NEXT();
+  }
+
+  // -------------------------------------------------------
+  // Inserts
+  // -------------------------------------------------------
+  OP(Insert) : {
+    auto table_id = frame->LocalAt<uint32_t>(READ_LOCAL_ID());
+    auto values = frame->LocalAt<byte *>(READ_LOCAL_ID());
+    auto exec_ctx = frame->LocalAt<exec::ExecutionContext *>(READ_LOCAL_ID());
+    OpInsert(exec_ctx, table_id, values);
+    EXECUTION_LOG_TRACE("Inserted into table ", table_id);
+    DISPATCH_NEXT();
+  }
+
+  // -------------------------------------------------------
+  // Index Iterator
+  // -------------------------------------------------------
+  OP(IndexIteratorConstruct) : {
```

Comment:
The other classes use the `Init` and `PerformInit` paradigm. It'd be less work to change this to `Init`.